parser/parser.y

```diff
@@ -4176,9 +4176,6 @@ CastType:
 	{
 		x := types.NewFieldType(mysql.TypeString)
 		x.Flen = $2.(int)
-		if $3.(bool) {
-			x.Flag |= mysql.BinaryFlag
-		}
 		x.Charset = $4.(string)
```

Comment:
The collate is not given by the syntax, the default collation will be used in runtime.